src/Yoti/Http/AbstractRequestHandler.php

```diff
@@ -98,61 +158,72 @@ public function getSdkId()
      */
     public function getPem()
     {
-        return $this->pem;
+        return (string) $this->pemFile;
     }
 
     /**
-     * Return the request headers including the signed message.
+     * Set SDK identifier.
      *
-     * @param string $signedMessage
+     * Allows plugins to declare their identifier.
      *
-     * @return array
+     * @param string $sdkIdentifier
+     *   SDK or Plugin identifier
+     *
+     * @throws RequestException
      */
-    private function generateRequestHeaders($signedMessage)
+    public function setSdkIdentifier($sdkIdentifier)
     {
-        // Prepare request Http Headers
-        $requestHeaders = [
-            CurlRequestHandler::YOTI_AUTH_HEADER_KEY . ": {$this->authKey}",
-            CurlRequestHandler::YOTI_DIGEST_HEADER_KEY . ": {$signedMessage}",
-            CurlRequestHandler::YOTI_SDK_IDENTIFIER_KEY . ": {$this->sdkIdentifier}",
-            'Content-Type: application/json',
-            'Accept: application/json',
-        ];
-
-        if ($version = Config::getInstance()->get('version')) {
-            $requestHeaders[] = self::YOTI_SDK_VERSION . ": {$this->sdkIdentifier}-{$version}";
+        if (!in_array($sdkIdentifier, $this->acceptedsdkIdentifiers, true)) {
+            throw new RequestException("Wrong Yoti SDK identifier provided: {$sdkIdentifier}");
```

Comment:
I'm not a fan of saying "Wrong" in error messages without saying why 🙂 
```suggestion
            throw new RequestException("{$sdkIdentifier} is not in the list of accepted identifiers");
```

Reply:
Fixed in 2a8784b